gui_interactions.md

```diff
@@ -0,0 +1,416 @@
+# GUI interactions
+
+This document is meant to inform and guide the development of various GUI
+interactions for Ignition Gazebo.
+
+## Overview
+
+Ignition Gazebo's graphical interface's high-level goals are to:
+
+1. **Introspect** a running simulation
+2. **Interact** with a running simulation
+3. **Edit** simulation worlds and robots
+
+**Introspection** allows users to see how objects are positioned through a 3D
+view, the precise position of objects through text widgets, how velocities
+are changing over time through plots, etc.
+
+**Interaction** allows users to apply forces to a robot while it moves,
+to teleport objects in the scene while simulation is running, etc.
+
+**Editing** allows users to arrange world environments, assemble new
+robots and objects, modify existing objects, etc.
+
+The main difference between interaction and editing is that the former must
+happen while the simulation is running, while the latter doesn't. But in several
+aspects, their functionalities overlap. For example, teleporting an object may
+be useful while the simulation is running to test whether a robot will keep
+following it. And teleporting objects is also essential to creating a new world
+environment.
+
+It could be argued that every aspect of editing could be helpful for
+interaction. Even actions like changing mass or adding a new link will be
+helpful to some user. However, it's important to keep in mind that handling
+interactions is considerably more complicated than handling editing, primarily
+because physics engines are usually designed to deal with nodies that have
```

Comment:
nodies -> nodes

Reply:
haha sorry, I had an uncommitted edit fixing this already. It was actually `bodies`:  4533188